### DISCUSSION TITLE
impl<R: Responder> Responder for MutexGuard<R>

### DISCUSSION THREAD

**Discussion by cubodix** [rwf2/Rocket#2615] [Questions] [answered]
i have a server with a page who updates every 10 seconds, so it is not 'static, such page is stored in a Mutex<Vec>, but if i pass the MutexGuard, rocket doesn't accept it, i think a implementation for this should be: use the inside, and unlock it

solution: implement MutexGuard for Responder, i think it should look like Box<R>, but with some modifications:
```rust
impl<'r, 'o: 'r, T: Responder<'r, 'o> + Sized> Responder<'r, 'o> for Box<T> {
    fn respond_to(self, req: &'r Request<'_>) -> response::Result<'o> {
        let inner = *self;
        inner.respond_to(req)
    }
}
```

extra: maybe also implement with RwLockReadGuard or such

**Comment by SergioBenitez** [comment, marked as answer]
It sounds like this is a problem with your approach, not functionality missing in Rocket. Your data probably _is_ `'static`, seeing as you're sharing it across the application, so this isn't part of the problem. Returning a `MutexGuard` from a handler would mean maintaining the lock until the response is completely written, which is prone to DoS issues (the client can effectively hold your lock as long as they want.) You probably want to find some way of avoiding trying to hold a lock across a handler invocation.

I'm not sure what you're doing, but perhaps if you gave an example we could be of more help. I'm converting this to a discussion.

**Reply by cubodix** [reply]
ut right i haven't through about my client locking until timeout
